timely_beliefs/beliefs/classes.py

```diff
@@ -544,8 +572,21 @@ def apply_belief_timing_filters(q):
                 ),
             )
 
+        # apply fast-track most recent event|belief only approach
+        if most_recent_event_only:
+            if most_recent_belief_only:
+                q = q.order_by(cls.event_start.desc(), cls.belief_horizon.desc()).limit(
+                    1
```

Comment:
I wonder whether we should make the `most_recent_event` option into a `n_most_recent_events` option. Could be more useful? For example, for retrieving the most recent day of data.

Reply:
It's an interesting thought.

But looking at designing it, it clashes a bit with existing ways to do that (where we already allow to express a time range). And I don't have the right use case for it at hand, either. Maybe later?

Reply:
It could work together with `event_ends_before` but not really with `event_starts_before` ...

Reply:
Clarification request:
- Did you mean `event_starts_before` or `event_starts_after`?
- Why would it not work together? We'd limit the time range, and within that time range attempt to obtain the right side up to some limited number of rows.

In practice, though, the relationship between the number of rows from the right side and the corresponding time span is not always clear. But it can be, provided that the user has set up their application to fill the sensor with data in such a way that they are guaranteed:
- a limited number of sources per event,
- a limited number of beliefs per event, and
- a limited probabilistic depth (rows per belief).

Let's say they have max 3 sources per event, max 12 beliefs per event and only deterministic beliefs, for a sensor with 1-hour resolution. Then they could query the most recent `3*12*24` rows to obtain a data set that *must* contain the most recent day.

Reply:
This sounds pretty technical to set up right (not all users would want to do this), and then it's also not really safe. Changing something in one part of your app (e.g. adding a data source, or the forecasting service adds more distinct beliefs) would change/break desired behavior in another (a dashboard that is using this feature).

Reply:
True. Now that we've discussed its usefulness, let's not pursue it further in this PR.